Model/Api/Insurance/InsuranceUpdate.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace Alma\MonthlyPayments\Model\Api\Insurance;
+
+use Alma\API\Exceptions\AlmaException;
+use Alma\MonthlyPayments\Api\Insurance\InsuranceUpdateInterface;
+use Alma\MonthlyPayments\Helpers\AlmaClient;
+use Alma\MonthlyPayments\Helpers\Functions;
+use Alma\MonthlyPayments\Helpers\InsuranceSubscriptionHelper;
+use Alma\MonthlyPayments\Helpers\Logger;
+use Alma\MonthlyPayments\Model\Insurance\ResourceModel\Subscription;
+use Magento\Backend\Model\Url;
+use Magento\Framework\Exception\AlreadyExistsException;
+use Magento\Framework\Notification\NotifierPool;
+use Magento\Framework\Webapi\Exception;
+use Magento\Framework\Webapi\Rest\Request;
+use Magento\Sales\Model\OrderRepository;
+
+class InsuranceUpdate implements InsuranceUpdateInterface
+{
+    /**
+     * @var Logger
+     */
+    private $logger;
+    /**
+     * @var Request
+     */
+    private $request;
+    private $almaClient;
+    private $insuranceSubscriptionHelper;
+    private $subscription;
+    private $notifierPool;
+    private $orderRepository;
+    private $url;
+
+    /**
+     * @param Request $request
+     * @param Logger $logger
+     */
+    public function __construct(
+        Request                     $request,
+        Logger                      $logger,
+        AlmaClient                  $almaClient,
+        InsuranceSubscriptionHelper $insuranceSubscriptionHelper,
+        Subscription                $subscription,
+        NotifierPool                $notifierPool,
+        OrderRepository             $orderRepository,
+        Url                         $url
+    )
+    {
+        $this->request = $request;
+        $this->logger = $logger;
+        $this->almaClient = $almaClient;
+        $this->subscription = $subscription;
+        $this->notifierPool = $notifierPool;
+        $this->orderRepository = $orderRepository;
+        $this->url = $url;
+        $this->insuranceSubscriptionHelper = $insuranceSubscriptionHelper;
+    }
+
+    /**
+     * @return string
+     * @throws Exception
+     */
+    public function update(): void
+    {
+        $params = $this->request->getParams();
+        $this->checkQeuryParams($params);
+
+        $subscriptionId = $params['subscription_id'];
+        $subscriptions = $this->getSubscription($subscriptionId);
```

Comment:
getSubscription or get Subscriptions? And what is exactly the object get with this function?